### PR TITLE
handle http proxy while generating routes

### PR DESCRIPTION
**Pull request by ramaraochavali**
PR https://github.com/istio/istio/pull/19160 added support for http_proxy protocol in side car config. That needs to be handled in routes. With out this, routes with invalid cluster like `outbound|0` are generated if a virtual service exists for a service and service does not have the port specified in the egress listener. See the new test cases added that explains the behaviour.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@rshriram @hzxuzhonghu PTAL

File: pilot/pkg/networking/core/v1alpha3/httproute_test.go [hunk 1]
```diff
@@ -417,7 +478,7 @@ func TestSidecarOutboundHTTPRouteConfig(t *testing.T) {
 			sidecarConfig:         sidecarConfig,
 			virtualServiceConfigs: nil,
 			expectedHosts: map[string]map[string]bool{
-				"test.com:8080": {"test.com:8080": true, "8.8.8.8:8080": true},
+				"test.com:8080": {"test.com": true, "test.com:8080": true, "8.8.8.8": true, "8.8.8.8:8080": true},
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/httproute_test.go]: Why this change?

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/httproute_test.go]: That is because earlier with out this change, this route is treated as http proxy route and we used to set listenerPort  = 0 and if it is zero, we merge https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/httproute.go#L346 here, which merged all the virtual hosts, Now with this change since it is not http_proxy, we do not merge - so we get all virtual hosts.

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/httproute_test.go]: IC, please update the test comments as well

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/httproute_test.go]: Sorry, Did not notice the test comments. I have updated now. PTAL
